web/api/web_api_v1.c

```diff
@@ -137,16 +137,25 @@ char *get_mgmt_api_key(void) {
 
         // save it
         fd = open(api_key_filename, O_WRONLY|O_CREAT|O_TRUNC, 444);
-        if(fd == -1)
-            fatal("Cannot create unique management API key file '%s'. Please fix this.", api_key_filename);
+        if(fd == -1) {
+            error("Cannot create unique management API key file '%s'. Please fix this.", api_key_filename);
+            goto temp_key;
+        }
 
-        if(write(fd, guid, GUID_LEN) != GUID_LEN)
-            fatal("Cannot write the unique management API key file '%s'. Please fix this.", api_key_filename);
+        if(write(fd, guid, GUID_LEN) != GUID_LEN) {
+            error("Cannot write the unique management API key file '%s'. Please fix this.", api_key_filename);
```

Comment:
A suggestion to help our users, instead `Please fix this`, I think we could say more explicit, saying for users that the permission of directory `netdata_configured_varlib_dir` does not allow to write something  or disk is full.

Reply:
Hey Thiago, thanks!

Indeed though, at that point we can't be sure if the whole `netdata_configured_varlib_dir` is problematic, or just the particular file/path (as set in `netdata management api key file`). I would also be hesitant to guess there what the reason might be for the failure (permissions, disk full?)

If `netdata_configured_varlib_dir` is a problem, then I think we should deal with it somewhere even from main. This place only deals with that particular key.

The dir pointed by the `netdata management api key file` setting will be printed in the output here. We could add a small piece of text to inform user what is needed to change, but then, should that be a trend with other errors as well?

We could e.g. change `Please fix this` to `Please configure 'netdata management api key file' parameter to a proper path and filename`. ?

Reply:
> The dir pointed by the netdata management api key file setting will be printed in the output here. We could add a small piece of text to inform user what is needed to change, but then, should that be a trend with other errors as well?

If we add the directory that is having problem, it will be clear for user where is the problem. Right now the current message says something about a file that could not be present on file system.

> We could e.g. change Please fix this to Please configure 'netdata management api key file' parameter to a proper path and filename. ?

I liked the last suggestion, because `Please fix it`, is not given any tip for users about possible problems they are having.